app/models/piece.rb

```diff
@@ -58,6 +60,13 @@ def obstructed?(new_x, new_y)
 
   private
 
+  def increment_move
+    game.update_attributes(move_number: game.move_number + 1)
+    update_attributes(game_move_number: game.move_number)
+    update_attributes(piece_move_number: piece_move_number + 1)
+    update_attributes(has_moved: true)
```

Comment:
Could combine 3 of these lines into 1 to reduce database queries.

`update_attributes(game_move_number: game.move_number, piece_move_number:...)`

Each call to `update_attributes` = 1 query